tss-esapi/src/abstraction/transient.rs

```diff
@@ -494,6 +421,117 @@ impl TransientKeyContext {
+            .with_decrypt(decrypt_flag)
+            .with_sign_encrypt(true)
+            .with_restricted(false)
+            .build()?;
+
+        let mut pub_builder = PublicBuilder::new()
+            .with_public_algorithm(match params {
+                KeyParams::Ecc { .. } => PublicAlgorithm::Ecc,
+                KeyParams::Rsa { .. } => PublicAlgorithm::Rsa,
+            })
+            .with_name_hashing_algorithm(HashingAlgorithm::Sha256)
+            .with_object_attributes(object_attributes);
+        match params {
+            KeyParams::Rsa {
+                size,
+                scheme,
+                pub_exponent,
+            } => {
+                let unique = match pub_key {
+                    None => PublicKeyRsa::default(),
+                    Some(PublicKey::Rsa(val)) => val.try_into()?,
+                    _ => return Err(Error::local_error(ErrorKind::InconsistentParams)),
+                };
+                pub_builder = pub_builder
+                    .with_rsa_parameters(
+                        PublicRsaParametersBuilder::new()
+                            .with_scheme(match scheme {
+                                RsaScheme::RsaSsa { .. } | RsaScheme::RsaPss { .. } => scheme,
+                                _ => RsaScheme::Null,
+                            })
+                            .with_key_bits(size)
+                            .with_exponent(pub_exponent)
+                            .with_is_signing_key(true)
+                            .with_is_decryption_key(decrypt_flag)
+                            .with_restricted(false)
+                            .build()?,
+                    )
+                    .with_rsa_unique_identifier(&unique);
+            }
+            KeyParams::Ecc { scheme, curve } => {
+                let unique = match pub_key {
+                    None => EccPoint::default(),
+                    Some(PublicKey::Ecc { x, y }) => EccPoint::new(x.try_into()?, y.try_into()?),
+                    _ => return Err(Error::local_error(ErrorKind::InconsistentParams)),
+                };
+                pub_builder = pub_builder
+                    .with_ecc_parameters(
+                        PublicEccParametersBuilder::new_unrestricted_signing_key(scheme, curve)
+                            .build()?,
+                    )
+                    .with_ecc_unique_identifier(&unique);
+            }
+        }
+        pub_builder.build()
+    }
+
+    /// Load a key into a TPM given its [KeyMaterial]
+    ///
+    /// If the key has only a public part, it is loaded accordingly in the Owner Hierarchy
+    fn load_key(
+        &mut self,
+        params: KeyParams,
+        material: KeyMaterial,
+        auth: Option<Auth>,
+    ) -> Result<KeyHandle> {
+        let public = TransientKeyContext::get_public_from_params(params, Some(material.public))?;
+
+        self.set_session_attrs()?;
+        let key_handle = match material.private.is_empty() {
```

Comment:
Why not make it a good ol' `if`?

Reply:
hmm, true, I think I started with a different approach and switched to this without thinking of `if`.

Reply:
Done